src/js/reducers/conversation-reducer.js

```diff
@@ -48,11 +48,12 @@ const replaceMessage = (messages, query, newMessage) => {
 };
 
 const isEqual = (messageA, messageB) => {
+    // _tempId is a property of messages sent by the appUser
     if (messageA._id && messageB._tempId && messageA._id === messageB._tempId) {
         return true;
     }
 
-    if (!messageA._id || !messageB._tempId) {
+    if (!messageA._id && !messageB._id || !messageA._id && !messageB._tempId) {
```

Comment:
maybe just add a paragraph about the logic in this function, I don't get the `if` on line 52 either :(